workers/unity/Assets/Fps/Scripts/SetupLogic/MobileWorkerConnectorBase.cs

```diff
@@ -0,0 +1,137 @@
+    {
+        private const string AuthPlayer = "Prefabs/MobileClient/Authoritative/Player";
+        private const string NonAuthPlayer = "Prefabs/MobileClient/NonAuthoritative/Player";
+
+        private const string Small = "small";
+        private const string Large = "large";
+
+        public int TargetFrameRate = 60;
+
+        protected GameObject levelInstance;
+
+        protected ConnectionController connectionController;
+
+        private List<TileEnabler> levelTiles = new List<TileEnabler>();
+        public List<TileEnabler> LevelTiles => levelTiles;
+
+        public string IpAddress { get; set; }
+
+        protected abstract string GetWorkerType();
+
+        protected virtual async void Start()
+        {
+            Application.targetFrameRate = TargetFrameRate;
+            await AttemptConnect();
+        }
+
+        protected async Task AttemptConnect()
+        {
+            await Connect(GetWorkerType(), new ForwardingDispatcher()).ConfigureAwait(false);
+        }
+
+        protected override string SelectDeploymentName(DeploymentList deployments)
+        {
+            // This could be replaced with a splash screen asking to select a deployment or some other user-defined logic.
+            return deployments.Deployments[0].DeploymentName;
+        }
+
+        protected override void HandleWorkerConnectionEstablished()
+        {
+            var world = Worker.World;
+
+            // Only take the Heartbeat from the PlayerLifecycleConfig Client Systems.
+            world.GetOrCreateManager<HandlePlayerHeartbeatRequestSystem>();
+
+            GameObjectRepresentationHelper.AddSystems(world);
+            var fallback = new GameObjectCreatorFromMetadata(Worker.WorkerType, Worker.Origin, Worker.LogDispatcher);
+
+            // Set the Worker gameObject to the ClientWorker so it can access PlayerCreater reader/writers
+            GameObjectCreationHelper.EnableStandardGameObjectCreation(
+                world,
+                new AdvancedEntityPipeline(Worker, AuthPlayer, NonAuthPlayer, fallback),
+                gameObject);
+
+            LoadWorld();
+        }
+
+        protected override void HandleWorkerConnectionFailure(string errorMessage)
+        {
+            connectionController.OnFailedToConnect();
+        }
+
+        public override void Dispose()
+        {
+            if (levelInstance != null)
+            {
+                Destroy(levelInstance);
+            }
+
+            base.Dispose();
+        }
+
+        protected bool GetWorldLayerCount(out int worldLayerCount)
+        {
+            var workerSystem = Worker.World.GetExistingManager<WorkerSystem>();
+            var worldSize = Worker.Connection.GetWorkerFlag("world_size");
+
+            switch (worldSize)
+            {
+                case Small:
+                    worldLayerCount = 4;
+                    break;
+                case Large:
+                    worldLayerCount = 24;
```

Comment:
This should not be hardcoded in a mobile specific file. Move to a Scriptable object.